client/src/annoMatrix/annoMatrix.ts

```diff
@@ -308,163 +319,145 @@ export default class AnnoMatrix {
+    category: Category,
+    unassignedCategory: string
+  ): Promise<AnnoMatrix>;
 
-    Will throw column does not exist or is not writable.
+  /*
+  Drop an entire writable column, eg a user-created obs annotation.  Typical use
+  is to provide the "Delete Category" implementation.  Returns the new AnnoMatrix.
+  Will throw if not a writable annotation.
 
-    Example:
+  Will throw column does not exist or is not writable.
 
-      dropObsColumn("old annotations") ->  AnnoMatrix
-    */
-    _subclassResponsibility();
-  }
+  Example:
 
-  // @ts-expect-error ts-migrate(6133) FIXME: 'colSchema' is declared but its value is never rea... Remove this comment to see the full error message
-  // eslint-disable-next-line class-methods-use-this, @typescript-eslint/no-unused-vars, @typescript-eslint/explicit-module-boundary-types -- make sure subclass implements
-  addObsColumn(colSchema, Ctor, value) {
-    /*
-    Add a new writable OBS annotation column, with the caller-specified schema, initial value
-    type and value.
+    dropObsColumn("old annotations") ->  AnnoMatrix
+  */
+  abstract dropObsColumn(col: string): AnnoMatrix;
 
-    Value may be any one of:
-      * an array of values
-      * a primitive type, including null or undefined.
-    If an array, length must be the same as 'this.nObs', and constructor must equal 'Ctor'.
-    If a primitive, 'Ctor' will be used to create the initial value, which will be filled
-    with 'value'.
+  /*
+  Add a new writable OBS annotation column, with the caller-specified schema, initial value
+  type and value.
 
-    Throws if the name specified in 'colSchema' duplicates an existing obs column.
+  Value may be any one of:
+    * an array of values
+    * a primitive type, including null or undefined.
+  If an array, length must be the same as 'this.nObs', and constructor must equal 'Ctor'.
+  If a primitive, 'Ctor' will be used to create the initial value, which will be filled
+  with 'value'.
 
-    Returns a new AnnoMatrix.
+  Throws if the name specified in 'colSchema' duplicates an existing obs column.
 
-    Examples:
+  Returns a new AnnoMatrix.
 
-      addObsColumn(
-        { name: "foo", type: "categorical", categories: "unassigned" },
-        Array,
-        "unassigned"
-      ) -> AnnoMatrix
+  Examples:
 
-    */
-    _subclassResponsibility();
-  }
+    addObsColumn(
+      { name: "foo", type: "categorical", categories: "unassigned" },
+      Array,
+      "unassigned"
+    ) -> AnnoMatrix
 
-  // @ts-expect-error ts-migrate(6133) FIXME: 'oldCol' is declared but its value is never read.
-  // eslint-disable-next-line class-methods-use-this, @typescript-eslint/no-unused-vars, @typescript-eslint/explicit-module-boundary-types -- make sure subclass implements
-  renameObsColumn(oldCol, newCol) {
-    /*
-    Rename the obs column 'oldCol' to have name 'newCol' and returns new AnnoMatrix.
+   */
+  abstract addObsColumn<T extends string | number | boolean | null | undefined>(
```

Comment:
I am not sure if I have typed the argument types correctly here for `addObsColumn` and would appreciate feedback on this!